clients/src/main/java/org/apache/kafka/clients/producer/RecordMetadata.java

```diff
@@ -51,6 +52,11 @@ private RecordMetadata(TopicPartition topicPartition, long offset, long timestam
         this.topicPartition = topicPartition;
     }
 
+    @Deprecated
+    public RecordMetadata(TopicPartition topicPartition, long baseOffset, long relativeOffset) {
+        this(topicPartition, baseOffset, relativeOffset, Record.NO_TIMESTAMP, -1, -1, -1);
```

Comment:
Thanks. Any chance of adding record.NO_CHECKSUM and record.NO_SIZE?

We have several places in the code where we call the constructor with -1 and I think this will improve readability.